docs/arch/adr-005-host-packages.md

```diff
@@ -0,0 +1,54 @@
+# ADR 5: Host Packages
+
+## Context
+
+Today, kURL bundles all host OS packages and their sub-dependencies in the add-on packages it generates and installs these packages in the end customer environment, rather than directly from the package repositories.
+
+For RPM based distributions, kURL sets up a local "kurl.local" repository and overrides the official OS mirrors for dependency resolution when installing these packages.
+Even worse, for APT based distributions, kURL uses the `dpkg` command rather than `apt`, forgoes dependency resolution, and force installs the packages, often breaking the dependency graph.
+
+This negatively affects the end customer, as dependency incompatibilities can result in broken OS packages, and often prevent upgrading the host OS when applying upgrades and security patches.
+Additionally, this is an unsupported means of installing, likely nullifying enterprise support contracts with the host OS provider.
+Finally, packaging an entirely new dependency chain for each host OS and version results in a significant amount of data (on the order of gigabytes) that must be downloaded and stored by the end customer each time they want to run a kURL install or upgrade.
+
+For us, the maintainers of kURL, this "feature" has its own set of challenges.
+Since host packages are targeted to OS and version, this requires additional work from the team to add support for each new OS and version.
+Additionally, as kURL is taking ownership of these OS packages, each supported OS must be added to the test matrix, adding significant time and cost to Testgrid.
+
+## Decision
+
+Rather than bundle host packages in the add-on archive, kURL will install the required packages using the official repos already available to the host.
+
+## Solution
+
+It is likely that the server already has access to official mirrors or satellite package repositories.
+If the packages are available in a repository, then kURL will install the packages from there.
+
+A set of preflights will be run at the start of installation to determine if any of the packages are not available.
+If not the script will exit early with a friendly message to the user detailing the missing packages, including a command to install these packages.
+
+kURL will treat add-ons that are host packages themselves a bit differently (Containerd, Docker, Kubernetes Kubelet, Collectd).
+These will continue to be bundled along with the add-on archive.
+Sub-dependencies will no longer be packaged along with the add-on archive and will be installed from remote repositories.
```

Comment:
Can or should we check to see if these packages are available and use them from the host if they are? I like that we're using the packaging system for su-dependencies.

Have you considered if we're shipping these that we statically compile them and avoid any host level sub-dependency?

Reply:
It makes sense to check for online installations. I will add that.

If there are statically compiled versions out there we can consider using them but I would not want to maintain our own. I suggest a proposal in an alternative ADR.

Reply:
Added a couple sentences about checking the hosted repo first.